src/mrinufft/_utils.py

```diff
@@ -4,6 +4,57 @@
 from collections import defaultdict
 from functools import wraps
 import numpy as np
+from numpy.typing import DTypeLike
```

Comment:
Maybe have a look at https://github.com/pearu/pydlpack
This is pretty much what you want to do right?

Reply:
welll the typing annotation is just fancy for now. 

From what I understood, all the array libraries have a `from_dlpack` so we don't need the python package you mention earlier. And we still need to handles the device <-> host communication 
```
>>> import numpy as np
>>> import cupy as cp 
>>> np.from_dlpack(cp.arange(10))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: Unsupported device in DLTensor.
```